spin fails to build on amd64

The package spin with version 3.2.0 and epoch 0 fails to build from source with the following error:

```
6  | / wasmtime::component::bindgen!({
7  | |     inline: r#"
8  | |     package fermyon:runtime;
9  | |     world host {
...  |
38 | |     trappable_imports: true,
39 | | });
   | |__^ the evaluated program panicked at 'assertion failed: 9 == <Method as wasmtime::component::ComponentType>::SIZE32', crates/world/src/lib.rs:6:1
   |
   = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of constant value failed
  --> crates/world/src/lib.rs:6:1
   |
6  | / wasmtime::component::bindgen!({
7  | |     inline: r#"
8  | |     package fermyon:runtime;
9  | |     world host {
...  |
38 | |     trappable_imports: true,
39 | | });
   | |__^ the evaluated program panicked at 'assertion failed: 9 == <Scheme as wasmtime::component::ComponentType>::SIZE32', crates/world/src/lib.rs:6:1
   |
   = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of constant value failed
  --> crates/world/src/lib.rs:6:1
   |
6  | / wasmtime::component::bindgen!({
7  | |     inline: r#"
8  | |     package fermyon:runtime;
9  | |     world host {
...  |
38 | |     trappable_imports: true,
39 | | });
   | |__^ the evaluated program panicked at 'assertion failed: 14 == <DnsErrorPayload as wasmtime::component::ComponentType>::SIZE32', crates/world/src/lib.rs:6:1
   |
   = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of constant value failed
  --> crates/world/src/lib.rs:6:1
   |
6  | / wasmtime::component::bindgen!({
7  | |     inline: r#"
8  | |     package fermyon:runtime;
9  | |     world host {
...  |
38 | |     trappable_imports: true,
39 | | });
   | |__^ the evaluated program panicked at 'assertion failed: 11 == <TlsAlertReceivedPayload as wasmtime::component::ComponentType>::SIZE32', crates/world/src/lib.rs:6:1
   |
   = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0080]: evaluation of constant value failed
  --> crates/world/src/lib.rs:6:1
   |
6  | / wasmtime::component::bindgen!({
7  | |     inline: r#"
8  | |     package fermyon:runtime;
9  | |     world host {
...  |
38 | |     trappable_imports: true,
39 | | });
   | |__^ the evaluated program panicked at 'assertion failed: 17 == <KeyResponse as wasmtime::component::ComponentType>::SIZE32', crates/world/src/lib.rs:6:1
   |
   = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0080`.
error[E0080]: evaluation of constant value failed
   --> /home/build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/wasmtime-wasi-29.0.1/src/bindings.rs:147:9
    |
147 | /         wasmtime::component::bindgen!({
148 | |             path: "wit",
149 | |             world: "wasi:cli/command",
150 | |             tracing: true,
...   |
180 | |             require_store_data_send: true,
181 | |         });
    | |__________^ the evaluated program panicked at 'assertion failed: 1 == <DirectoryEntry as wasmtime::component::ComponentType>::ALIGN32', /home/build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/wasmtime-wasi-29.0.1/src/bindings.rs:147:9
    |
    = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `spin-world` (lib) due to 46 previous errors
warning: build failed, waiting for other jobs to finish...
error[E0080]: evaluation of constant value failed
   --> /home/build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/wasmtime-wasi-29.0.1/src/bindings.rs:327:5
    |
327 | /     wasmtime::component::bindgen!({
328 | |         path: "wit",
329 | |         world: "wasi:cli/command",
330 | |         tracing: true,
...   |
416 | |         },
417 | |     });
    | |______^ the evaluated program panicked at 'assertion failed: 1 == <DirectoryEntry as wasmtime::component::ComponentType>::ALIGN32', /home/build/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/wasmtime-wasi-29.0.1/src/bindings.rs:327:5
    |
    = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `wasmtime-wasi` (lib) due to 2 previous errors
failed to build package: unable to run package spin pipeline: unable to run pipeline: exit status 101
```

--- a/spin.yaml
+++ b/spin.yaml
@@ -1,7 +1,7 @@
 package:
   name: spin
   version: "3.2.0"
-  epoch: 0
+  epoch: 1
   description: "Spin is the open source developer tool for building and running serverless applications powered by WebAssembly."
   copyright:
     - license: Apache-2.0
@@ -36,13 +36,17 @@ pipeline:
 
   - name: Configure and build
     runs: |
-      # This build requires the stable version of rust, managed by rustup, because it requires a few other toolchains too.
-      rustup install stable
+      # This build requires other toolchains so we use rustup.
+      #
+      # TODO: We're pinned at 1.85.0 because certain WASM dependencies don't
+      # compile with 1.86.0. Revert to `rustup install stable' when upstream
+      # release (they've already fixed this upstream)
+      rustup install 1.85.0
       rustup target add wasm32-wasip1
       rustup target add wasm32-unknown-unknown
 
-      # This is a bit of a hack, but it's the easiest way to get the right version of rustc and cargo in the path.
-      export PATH="$HOME/.rustup/toolchains/stable-${{build.arch}}-unknown-linux-gnu/bin:$PATH"
+      # Put our rustup-installed toolchains into $PATH
+      export PATH="$(dirname "$(rustup which cargo)"):$PATH"
 
       if [ "${{build.arch}}" = "aarch64" ]; then
           echo '[build]' >> .cargo/config
@@ -58,6 +62,7 @@ pipeline:
 
 update:
   enabled: true
+  manual: true # See TODO in "Configure and build" above
   github:
     identifier: fermyon/spin
     strip-prefix: v

